src/AltinnCore/Runtime/Controllers/PartiesController.cs

```diff
@@ -2,55 +2,133 @@
 
 namespace AltinnCore.Runtime.Controllers
 {
     /// <summary>
     /// Handles party related operations
     /// </summary>
     [Authorize]
-    public class PartiesController : Controller
+    [ApiController]
+    public class PartiesController : ControllerBase
     {
         private readonly ILogger _logger;
         private readonly IAuthorization _authorization;
         private readonly UserHelper _userHelper;
+        private readonly IRepository _repository;
+        private readonly IProfile _profile;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PartiesController"/> class
         /// <param name="logger">The logger</param>
         /// <param name="authorization">the authorization service handler</param>
         /// <param name="profileService">the profile service</param>
         /// <param name="registerService">the register service</param>
-        /// <param name="userHelper">the user helper</param>
+        /// <param name="repository">The repository service</param>
         /// </summary>
         public PartiesController(
                     ILogger<PartiesController> logger,
                     IAuthorization authorization,
                     IProfile profileService,
-                    IRegister registerService)
+                    IRegister registerService,
+                    IRepository repository)
         {
             _logger = logger;
             _authorization = authorization;
             _userHelper = new UserHelper(profileService, registerService);
+            _repository = repository;
+            _profile = profileService;
         }
 
         /// <summary>
-        /// Gets the lsit of parties the user can represent
+        /// Gets the list of parties the user can represent
         /// </summary>
         /// <returns></returns>
-        [HttpGet]
+        [HttpGet("/Get")]
         public IActionResult Get()
         {
             UserContext userContext = _userHelper.GetUserContext(HttpContext).Result;
             List<Party> partyList = _authorization.GetPartyList(userContext.UserId);
             return Ok(partyList);
         }
+
+        /// <summary>
+        /// Validates party and profile settings before the end user is allowed to instantiate a new app instance
+        /// </summary>
+        /// <param name="org">The organization<param>
+        /// <param name="app">The application<param>
+        /// <param name="partyId">The selected partyId<param>
+        /// <returns>A validation status</returns>
+        [HttpPost("{org}/{app}/api/v1/parties/validateInstantiation")]
+        public IActionResult ValidateInstantiation(string org, string app, [FromQuery] int partyId)
+        {
+            UserContext userContext = _userHelper.GetUserContext(HttpContext).Result;
+            UserProfile user = _profile.GetUserProfile(userContext.UserId).Result;
+            List<Party> partyList = _authorization.GetPartyList(userContext.UserId);
```

Comment:
If partyList is null there is no need to continue further down.. Maybe add a check for this and a return.

Reply:
Unless the user itself will always be present in this list?

Reply:
The partyList may be null, but the user can still represent itself.